relay-server/src/utils/memory.rs

```diff
@@ -83,6 +83,17 @@ impl MemoryStat {
         **self.0.memory.load()
     }
 
+    /// Returns a copy of the memory data at this point in time.
+    ///
+    /// Unlike [`Self::memory`], this method unconditionally refreshes the memory state
+    /// and returns the most up-to-date readings. Use it when you need an accurate
+    /// snapshot of memory at the exact time of the call. Avoid using it unless
+    /// strictly necessary, as it may impact performance.
+    pub fn current_memory(&self) -> Memory {
```

Comment:
As discussed, we can make this into a static method which doesn't need all the refresh debounce logic. `Memory::current_memory()` instead of `Memory::new(0).current_memory()`.